File: ISA-CMOP_Python/features/f_skew.py
from scipy.stats import skew
from features.feature_helpers import remove_imag_rows
import numpy as np


def f_skew(pop):
    """
    Checks the skewness of the population of objective values - only univariate avg/max/min/range. Removed rank skew computation since it was commented out in Alsouly's code.
    """
    obj = pop.extract_obj()
    obj = remove_imag_rows(obj)

    if obj.size > 0:
        skew_avg = np.mean(skew(obj, axis=0))
        skew_max = np.max(skew(obj, axis=0))
        skew_min = np.min(skew(obj, axis=0))
        skew_rnge = skew_max - skew_min
    return [skew_avg, skew_min, skew_max, skew_rnge]

File: ISA-CMOP_Python/testing/f_skew.m
function [skew_avg, skew_min, skew_max, skew_rnge] = f_skew(objvar)
%% Checks the skewness of the population of objective values - both the rank
    % and univariate avg/max/min/range
	rmimg = find(sum(imag(objvar)~= 0,2));
	objvar(rmimg,:) = [];

    %ranksort = NDSort(objvar, length(objvar));
    [skew_avg, skew_min, skew_max, skew_rnge] = deal(NaN);
    if numel(objvar) > 0
        %skew_rank = skewness(ranksort);
        skew_avg = mean(skewness(objvar));
        skew_max = max(skewness(objvar));
        skew_min = min(skewness(objvar));
        skew_rnge = skew_max - skew_min;
    end
end

